lib/libyul2llvm/YulASTVisitor/CodegenVisitor.cpp

```diff
@@ -567,9 +569,18 @@ llvm::StructType *LLVMCodegenVisitor::getExtCallCtxType() {
   return extCallCtxType;
 }
 
+void LLVMCodegenVisitor::setSelfPointer(llvm::Value *v) { selfPointer = v; }
+
 llvm::Value *LLVMCodegenVisitor::getSelfArg() const {
   assert(currentFunction && "currentFunction null while getting silfArg");
-  return currentFunction->arg_begin();
+  if (!isConstructor(currentFunction->getName()))
+    return currentFunction->arg_begin();
+  else if (selfPointer != nullptr)
+    return selfPointer;
+  else {
+    llvm::errs() << "Self pointer is null";
```

Comment:
nit: this is more helpful:
```suggestion
    llvm::errs() << "Self pointer is null in constructor";
```

Reply:
done